deployment/terraform/modules/vpc/main.tf

```diff
@@ -82,3 +82,70 @@ resource "aws_route_table_association" "private_subnet" {
   subnet_id      = "${aws_subnet.private_subnet_a.id}"
   route_table_id = "${aws_route_table.private_route_table.id}"
 }
+
+resource "aws_security_group" "elb" {
+  # using standalone security group rule resources to avoid cycle errors
+  description = "Refinery: allow HTTP/S ingress and HTTP egress"
+  name        = "${terraform.workspace}-elb"
+  tags        = "${var.tags}"
+  vpc_id      = "${aws_vpc.vpc.id}"
+}
+
+resource "aws_security_group_rule" "http_ingress" {
+  description       = "Refinery: allow HTTP ingress from Internet to ELB"
+  type              = "ingress"
+  cidr_blocks       = ["0.0.0.0/0"]
+  from_port         = 80
+  to_port           = 80
+  protocol          = "tcp"
+  security_group_id = "${aws_security_group.elb.id}"
+}
+
+resource "aws_security_group_rule" "https_ingress" {
+  description       = "Refinery: allow HTTPS ingress from Internet to ELB"
+  type              = "ingress"
+  cidr_blocks       = ["0.0.0.0/0"]
+  from_port         = 443
+  to_port           = 443
+  protocol          = "tcp"
+  security_group_id = "${aws_security_group.elb.id}"
+}
+
+resource "aws_security_group_rule" "http_egress" {
+  description              = "Refinery: allow HTTP egress from ELB to app server"
+  type                     = "egress"
+  from_port                = 80
+  to_port                  = 80
+  protocol                 = "tcp"
+  security_group_id        = "${aws_security_group.elb.id}"
+  source_security_group_id = "${aws_security_group.app_server.id}"
```

Comment:
Double check this: In the original, it's **Destination**SecurityGroupId.
```
    cft.resources.elbegress = core.Resource(
        'ELBEgress', 'AWS::EC2::SecurityGroupEgress',
        core.Properties({
            "GroupId": functions.get_att('ELBSecurityGroup', 'GroupId'),
            "IpProtocol": "tcp",
            "FromPort": "80",
            "ToPort": "80",
            "DestinationSecurityGroupId": functions.get_att(
                'InstanceSecurityGroup', 'GroupId'),
        })
    )
```

Reply:
Yeah, the name is odd but works as advertised:
[source_security_group_id](https://www.terraform.io/docs/providers/aws/r/security_group_rule.html#source_security_group_id) - (Optional) The security group id to allow access to/from, depending on the type.